src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/EnvelopeProcessor.java

```diff
@@ -1,68 +1,35 @@
 package uk.gov.hmcts.reform.bulkscan.orchestrator.services;
 
+import com.microsoft.azure.servicebus.ExceptionPhase;
 import com.microsoft.azure.servicebus.IMessage;
-import com.microsoft.azure.servicebus.IMessageReceiver;
-import com.microsoft.azure.servicebus.primitives.ServiceBusException;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
-import org.springframework.scheduling.annotation.Scheduled;
-import org.springframework.stereotype.Service;
+import com.microsoft.azure.servicebus.IMessageHandler;
+import org.springframework.stereotype.Component;
 import uk.gov.hmcts.reform.bulkscan.orchestrator.services.bulkscanprocessorclient.client.BulkScanProcessorClient;
-import uk.gov.hmcts.reform.bulkscan.orchestrator.services.bulkscanprocessorclient.exceptions.ReadEnvelopeException;
-import uk.gov.hmcts.reform.bulkscan.orchestrator.services.bulkscanprocessorclient.model.Envelope;
-import uk.gov.hmcts.reform.bulkscan.orchestrator.services.servicebus.ReceiverProvider;
-import uk.gov.hmcts.reform.bulkscan.orchestrator.services.servicebus.exceptions.ConnectionException;
 
-import java.util.Objects;
+import java.util.concurrent.CompletableFuture;
 
-@Service
-public class EnvelopeProcessor {
-    public static final String TEST_MSG_LABEL = "test";
+@Component
+public class EnvelopeProcessor implements IMessageHandler {
+    private BulkScanProcessorClient bulkScanProcessorClient;
 
-    private static final Logger logger = LoggerFactory.getLogger(EnvelopeProcessor.class);
-
-    private final ReceiverProvider receiverProvider;
-    private final BulkScanProcessorClient bulkScanProcessorClient;
-
-    public EnvelopeProcessor(
-        ReceiverProvider receiverProvider,
-        BulkScanProcessorClient bulkScanProcessorClient
-    ) {
-        this.receiverProvider = receiverProvider;
+    public EnvelopeProcessor(BulkScanProcessorClient bulkScanProcessorClient) {
         this.bulkScanProcessorClient = bulkScanProcessorClient;
     }
 
-    @Scheduled(fixedDelayString = "${queue.read-interval}")
-    public void run() {
-        logger.info("Checking queue for new messages");
-        try {
-            IMessageReceiver msgReceiver = receiverProvider.get();
-            IMessage msg = msgReceiver.receive();
-            while (msg != null) {
-                handle(msgReceiver, msg);
-                msg = msgReceiver.receive();
-            }
-        } catch (ServiceBusException | ConnectionException exc) {
-            logger.error("Unable to read messages from queue", exc);
-        } catch (InterruptedException exc) {
-            Thread.currentThread().interrupt();
-            logger.error("Unable to read messages from queue", exc);
-        }
+    @Override
+    public CompletableFuture<Void> onMessageAsync(IMessage message) {
+        CompletableFuture<Void> future = new CompletableFuture<>();
+        process(message);
+        future.complete(null);
```

Comment:
does `complete` have to be specifically called? i would've thought using this handler callback should've solved `CompletableFuture` .. oh wait! `process(message)` is not included into future at all. i think it should be so the benefit of concurrency design is achieved